main.js

```diff
@@ -1,3 +1,29 @@
 import './style.css';
 
-console.log('Hello world');
+const modalButton = document.getElementById('modal-button');
+const modal = document.getElementById('modal');
+const modalCloseButton = document.getElementById('modal-close-button');
+const toolTipButton = document.getElementById('tooltip-button');
+const toolTipText = document.getElementById('tooltip');
+
+modalButton.addEventListener('click', () => {
+    modal.style.display = 'block';
+});
+
+modalCloseButton.addEventListener('click', () => {
+    modal.style.display = 'none';
+});
+
+toolTipButton.addEventListener('click', () => {
+    if (toolTipText.style.visibility === 'visible') {
+        toolTipText.style.visibility = 'hidden'
```

Comment:
Visibility hide element for user but not from CCSOM. It is better to use `display: none` for managing visibility of objects like tooltip